zebra-state/src/service/finalized_state/zebra_db/block.rs

```diff
@@ -421,6 +421,14 @@ impl DiskWriteBatch {
         note_commitment_trees: NoteCommitmentTrees,
         value_pool: ValueBalance<NonNegative>,
     ) -> Result<(), BoxError> {
+        // Return a fake error for some tests.
+        #[cfg(test)]
+        {
+            if finalized.height < Height(100) && finalized.hash == block::Hash([0; 32]) {
```

Comment:
The condition is checked for every test we run so just I wanted to restrict the set of blocks that could trigger the failure in case someone adds future tests that would be committing blocks with hashes equal to zero.

Reply:
Let me add that as a comment.

Reply:
Done in [b75f85e](https://github.com/ZcashFoundation/zebra/pull/5377/commits/b75f85ec73681837b2b6f7183209609a7d0f743a).

Reply:
> The condition is checked for every test we run so just I wanted to restrict the set of blocks that could trigger the failure in case someone adds future tests that would be committing blocks with hashes equal to zero.

Ah. that makes sense. Thank you for adding the comment.